### PR TITLE
Kindle suspend using lipc

### PR DESCRIPTION
**Pull request by Galunid**
fixes some stuff discussed in #4823

### CODE REVIEW

File: frontend/apps/reader/modules/readergesture.lua [hunk 1]
```diff
@@ -783,7 +783,11 @@ function ReaderGesture:gestureAction(action, ges)
             })
         end
     elseif action == "suspend" then
-        UIManager:suspend()
+        if Device:isKindle() then
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readergesture.lua]: I assume this predates your other fix? The changes to this file are redundant and wouldn't belong here even if they weren't. ;-)

File: frontend/ui/uimanager.lua [hunk 2]
```diff
@@ -1109,7 +1109,7 @@ function UIManager:suspend()
     if Device:isCervantes() or Device:isKobo() or Device:isSDL() or Device:isSonyPRSTUX() then
         self.event_handlers["Suspend"]()
     elseif Device:isKindle() then
-        self.event_handlers["IntoSS"]()
+        Device.powerd:toggleSuspend(1)
```
Comment (Frenzie) [inline frontend/ui/uimanager.lua]: Is the idea that this power button press emulation will trigger an `IntoSS` event when that's handled in `frontend/device/input`?

File: frontend/device/kindle/powerd.lua [hunk 3]
```diff
@@ -89,4 +89,10 @@ function KindlePowerD:afterResume()
     end
 end
 
+function KindlePowerD:toggleSuspend(status)
+    if self.lipc_handle then
+        self.lipc_handle:set_int_property("com.lab126.powerd", "powerButton", status)
```
Comment (Frenzie) [inline frontend/device/kindle/powerd.lua]: lgtm, but @NiLuJe actually knows things about lipc

File: frontend/apps/reader/modules/readergesture.lua (hunk 1 above)
Comment (Galunid) [reply frontend/apps/reader/modules/readergesture.lua]: Oh yes, I accidentally added this, I'll remove it in a moment

File: frontend/ui/uimanager.lua (hunk 2 above)
Comment (Galunid) [reply frontend/ui/uimanager.lua]: Tbh I'm not certain, the idea is that power button suspends the device properly, so instead of creating such fixes, it directly emulates "power button press", thus triggering events related to this. It works normally and I don't think there could be any problems related to this. It is also what @NiLuJe suggested here https://github.com/koreader/koreader/issues/4823#issuecomment-474962797
It seems to work pretty well too.

**Comment by Galunid** [comment]
@Frenzie It should be labeled as Kindle, not Kobo :wink:

**Comment by Frenzie** [comment]
@Galunid I must've misclicked.

File: frontend/device/kindle/powerd.lua (hunk 3 above)
Comment (NiLuJe) [reply frontend/device/kindle/powerd.lua]: I actually haven't checked what passing 0 to it does. I think it'll do nothing, but I'll have to check.

Setting it to 1 does work when it's "asleep" for a wakeup, though.

TL;DR: This should probably always set it to 1 without accepting an input variable.

File: frontend/ui/uimanager.lua (hunk 2 above)
Comment (NiLuJe) [reply frontend/ui/uimanager.lua]: Yep, it'll fire the original event that IntoSS responds to.

File: frontend/device/kindle/powerd.lua (hunk 3 above)
Comment (Galunid) [reply frontend/device/kindle/powerd.lua]: Yes, I just checked, passing 0 does emulate button too
